android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt

```diff
@@ -44,11 +48,12 @@ open class Node<V : RibView>(
         internal const val KEY_ROUTER = "node.router"
         internal const val KEY_INTERACTOR = "node.interactor"
         internal const val KEY_VIEW_STATE = "view.state"
-        private val requestCodeRegistry = RequestCodeRegistry(8)
+        private val requestCodeRegistry = RequestCodeRegistry()
     }
 
     init {
         router.node = this
+        interactor.node = this
```

Comment:
Lint warning: Leaking 'this' in constructor of non-final class Node.
Don't you think that can be a potential danger?

Code smell. Node class is open only for internal test purposes. I think it's better to find another way to write tests and make the class final.

Reply:
This is true or false regardless of the current ticket (applies to line `router.node = this` as well). If you think it's serious and should be improved, please create a ticket for it.